pkg/booleanpolicy/policyversion/downgrade_test.go

```diff
@@ -143,26 +66,24 @@ func TestOrganizeByVersionRank(t *testing.T) {
 		{
 			"Single downgrader",
 			map[string]downgrader{
-				version1: dummy,
+				version1_1: dummy,
 			},
 			[]downgrader{
-				nil, dummy, nil,
+				dummy, nil, nil,
 			},
 		},
 		{
 			"All downgraders",
 			map[string]downgrader{
-				legacyVersion: dummy,
-				version1:      dummy,
-				version1_1:    dummy,
+				version1_1: dummy,
 			},
 			[]downgrader{
 				dummy, dummy, dummy,
 			},
 		},
 	}
 
-	versions := []string{legacyVersion, version1, version1_1}
+	versions := []string{version1_1}
```

Comment:
I do not have a crystal ball for the future :-) though I wish I did. My goal is to remove unnecessary, unused code paths and reduce code bloat.  The workflows team has to navigate this day in an day out, as we develop our features, unnecessarily.

Reply:
> In its current state, this PR removes _parts_ of the versioning functionality (with _some_ tests) rendering the whole feature useless but still present in the code. This is suboptimal, I suggest to take one of these 3 options:
> 
> 1. remove versioning in backend entirely, including all upgrade, downgrade, version compare code and tests;
> 2. same as ^ but also remove `Policy.policy_version` proto field;
> 3. remove the possibility to downgrade from current `1.1` version to anything below but keep all the versioning functionality for future need.
> 
> My intuition says that since we've already seen 2 cases where we needed to version our policies, it's hard to believe that no more cases will arise, so 3. seems to be the best option. However, I'm not familiar with the vision for our policy engine and hence might be off.
> 
> Let's agree (and capture it in the PR description) on what we want here before we resume reviewing.

Agree. 3 has my vote. Remove the complexity, and unnecessary baggage once and for all :-). My latest commit is towards that, check it out when you have a few!

Reply:
I'm confused. You say 3 but describe 1. What shall I review?

Reply:
Removed the downgrade path (please check latest commit when you get a chance).